ts/Stock/RangeSelector/RangeSelector.ts

```diff
@@ -364,9 +364,16 @@ class RangeSelector {
 
         // Update the chart
         if (!baseAxis) {
+            // X-axis options not defined, avoids undefined error and
+            // sets deferred range to undefined (#20529).
+            if (!chart.userOptions.xAxis) {
+                chart.userOptions.xAxis = {};
+            }
```

Comment:
As a rule of thumb we avoid checking `userOptions`, because we expect `options` to be correctly updated, regardless of the option in question comes from the user or the defaults.  Another rule of thumb is to not modify user options, because, well, then they are no longer user options.

Do we have another way of solving this? What happened on line 369 with the old logic? Did the splat result in `undefined`? In that case, could we do something like this?

`baseXAxisOptions = splat(chart.options.xAxis || {})[0];`

Reply:
To keep it short, when I started the task:
- The `chart.options` for some reason wasn't updated with the rangeSelection options and any attempts I made at updating them either had no effect or only broke multiple other tests. Currently, it seems rangeSelector options has been worked on separately by someone else in a separate PR as `options` are now properly updated before the range selection code runs, as far as I can see.
- I presumed, by your refactor (roughly a year ago, see bisected commit) which swapped `options` for `userOptions` in `Chart.getAxes()`, that `userOptions` was to be used, as there had to be a reason for why you refactored `getAxes` to use `userOptions`.

So, if `options` is to be used and not `userOptions`, should I change both of `RangeSelector.clickButton()` and `Chart.getAxes()` back to only use `options`? and of course fix the tests that fail when doing so?

Reply:
@Xillez The `Chart.getAxes` were part of a bigger refactor where the end game was to make sure that default options and user options were more cleanly separated. This happens at an earlier stage before the `chart.options` are reliably created, therefore we use `userOptions`.

You should not change `Chart.getAxis`, only `RangeSelector.clickButton` if that works.